pkg/gadgets/internal/socketenricher/tracer_test.go

```diff
@@ -179,56 +180,84 @@ func TestSocketEnricherBind(t *testing.T) {
 			t.Cleanup(tracer.Close)
 
 			var port uint16
+			var fd int
 
 			utilstest.RunWithRunner(t, runner, func() error {
 				var err error
-				port, err = test.generateEvent()
+				port, fd, err = test.generateEvent()
+
+				t.Cleanup(func() {
+					// cleanup only if it has not been already closed
+					if fd != -1 {
+						unix.Close(fd)
+					}
+				})
+
 				return err
 			})
 
-			iter := tracer.SocketsMap().Iterate()
-			var key socketenricherSocketsKey
-			var value socketenricherSocketsValue
-			var entries []socketEnricherMapEntry
-			for iter.Next(&key, &value) {
-				entries = append(entries, socketEnricherMapEntry{
-					Key:   key,
-					Value: value,
-				})
-			}
-			if err := iter.Err(); err != nil {
-				t.Fatal("Cannot iterate over socket enricher map:", err)
+			entries := socketsMapEntries(t, tracer, nil)
+
+			utilstest.ExpectAtLeastOneEvent(test.expectedEvent)(t, runner.Info, port, entries)
+
+			if fd != -1 {
+				unix.Close(fd)
+				// Disable t.Cleanup() above
+				fd = -1
 			}
 
-			if test.validateEvent != nil {
-				test.validateEvent(t, runner.Info, port, entries)
+			expected := test.expectedEvent(runner.Info, port)
+			entries = socketsMapEntries(t, tracer, func(e *socketEnricherMapEntry) bool {
+				return !reflect.DeepEqual(expected, e)
+			})
+			if len(entries) != 0 {
+				t.Fatalf("Entry not cleaned properly: %+v", entries)
 			}
 		})
 	}
 }
 
+func socketsMapEntries(t *testing.T, tracer *SocketEnricher, filter func(*socketEnricherMapEntry) bool) (entries []socketEnricherMapEntry) {
+	iter := tracer.SocketsMap().Iterate()
+	var key socketenricherSocketsKey
+	var value socketenricherSocketsValue
+	for iter.Next(&key, &value) {
+		entry := socketEnricherMapEntry{
+			Key:   key,
+			Value: value,
+		}
+		if filter != nil && filter(&entry) {
+			continue
+		}
+		entries = append(entries, entry)
+	}
+	if err := iter.Err(); err != nil {
+		t.Fatal("Cannot iterate over socket enricher map:", err)
+	}
+	return entries
+}
+
```

Comment:
What about implementing a `ExpectNoEventToMatch()` in helpers.go? I think that function can be useful for other testing as well.

Reply:
I'll try to implement that in a separate PR.